pkg/ibmvpc/parser.go

```diff
@@ -1486,71 +1447,100 @@ func getLoadBalancerIPs(vpcConfig *vpcmodel.VPCConfig,
 		if err != nil {
 			return nil, err
 		}
-		// first get name, id, address, publicAddress:
-		var name, id, address, publicAddress string
-		pipIndex, original := subnetsWithPrivateIPs[subnet]
-		if original {
-			// subnet has a private IP, we take it from the config
-			pIP := loadBalancerObj.PrivateIps[pipIndex]
-			name, id, address = *pIP.Name, *pIP.ID, *pIP.Address
-			if hasPublicAddress {
-				publicAddress = *loadBalancerObj.PublicIps[pipIndex].Address
+		subnetBlocks := subnetsBlocks.subnetBlocks(*subnetObj.CRN)
+		privateIPAddressesMessage := make([]string, len(subnetBlocks))
+		// we want a pip for every block of every subnet:
+		for blockIndex, subnetBlock := range subnetBlocks {
+			// first get name, id, address, publicAddress:
+			var name, id, address, publicAddress string
+			blockHasPrivateIP := subnetsPIPsAddresses[subnet] != nil &&
+				subnetsPIPsAddresses[subnet].ContainedIn(subnetBlock)
+			switch {
+			case blockHasPrivateIP:
+				// subnet block has a private IP, we take it from the config
+				pIP := loadBalancerObj.PrivateIps[subnetsPIPsIndexes[subnet]]
+				name, id, address = *pIP.Name, *pIP.ID, *pIP.Address
+				if hasPublicAddress {
+					publicAddress = *loadBalancerObj.PublicIps[subnetsPIPsIndexes[subnet]].Address
+				}
+			case subnetsBlocks.isFullyReservedBlock(*subnetObj.CRN, blockIndex):
```

Comment:
shouldn't we at least issue a warning here? or take a note? 
also, this implies that if, say, a vsi is deleted and its address is frees the connectivity of the LB increases. A bit fishy, isn't it? @adisos add to lint?

Reply:
the meaning here that user will get a warning when he set a specific address on an sg rule.

Reply:
Please add a explain this here in the continue context.

Reply:
done

Reply:
Still not clear enough. Lets fix this comment together, make sure it is clear.

Reply:
fixed together